BUILD.md

```diff
@@ -26,6 +26,39 @@ If you have SGX-enabled hardware (activated in BIOS, and with SGX kernel module
 you can use `./mob prompt --hw` to get SGX in the container. Then you can both build and
 run the tests in `SGX_MODE=HW`. (See below for an explanation.)
 
+##### IDE Support using the docker environment
+
+You can use `./mob prompt --cmd <command>` to run `<command>` from within
```

Comment:
How about we make a `run` action instead? e.g. `mob run cargo check ...`

Reply:
I wanted to, but due to limitations with argparse, I couldn't get it to work reliably. Depending on how it was called, in intuitive ways, it would conflict with the `container_name` positional argument, causing it to think `cargo` is the container name, for example, and then fail. Thats why I went with an argument instead.

It might be possible to work around this by using another library, or possibly more argparse features like subparsers/subcommands, but that would be a bigger change.

Reply:
I wanted to, but due to limitations with argparse, I couldn't get it to work reliably. Depending on how it was called, in intuitive ways, it would conflict with the `container_name` positional argument, causing it to think `cargo` is the container name, for example, and then fail. Thats why I went with an argument instead.

It might be possible to work around this by using another library, or possibly more argparse features like subparsers/subcommands, but that would be a bigger change.

Reply:
For example trying to use `parser.add_argument("cmd", nargs=argparse.REMAINDER)` and calling it as so `./mob prompt cargo check` results in the following arguments. It also does not work with `nargs="*"`

```python
Namespace(action='prompt', container_name='cargo', hw=False, ias_prod=False, dry_run=False, tag=None, no_pull=False, verbose=False, ssh_dir=False, ssh_agent=False, expose=None, publish=None, cmd=['check'])
```

Argparse just doesnt seem to properly support the bash/linux/posix? convention of `--`([see here](https://unix.stackexchange.com/questions/11376/what-does-double-dash-mean)), so it can't be used to signify the end of options in that manner.

Reply:
I see. Would you mind trying the `run` action and removing `nargs` param for the `add_argument("container_name", ...)` call (line 95)? `nargs='?', default=None` doesn't really make sense.

Reply:
On further thought, I guess that won't keep the `container_name` arg from slurping the first word. It seems more usable to have the Python code treat `args.container_name` as the first part of the command in `run` mode, but I won't block approval on that.

Reply:
That would mean its impossible to provide both arguments though. I'm not sure if it even makes sense to want to `run` with another `container_name`, though.